pkg/karmadactl/join/join.go

```diff
@@ -165,6 +180,11 @@ func (j *CommandJoinOption) Run(f cmdutil.Factory) error {
 			*options.DefaultConfigFlags.Context, *options.DefaultConfigFlags.KubeConfig, err)
 	}
 
+	// Configure impersonation
```

Comment:
Is there a scenario where field `.Impersonate.Extra` needs to be set?

Reply:
yeah might be useful, I did not have any use for it (neither for UID)
can be a simple later PR if anyone needs it
... also willing to add if anyone wants it

Reply:
We can wait to add it if we need to.